src/deva/utils/vault.py

```diff
@@ -0,0 +1,95 @@
+# SPDX-FileCopyrightText: 2024-present Datadog, Inc. <dev@datadoghq.com>
+#
+# SPDX-License-Identifier: MIT
+import urllib.parse
+import webbrowser
+
+import hvac
+
+OIDC_CALLBACK_PORT = 8250
+OIDC_REDIRECT_URI = f"http://localhost:{OIDC_CALLBACK_PORT}/oidc/callback"
+SELF_CLOSING_PAGE = """
+<!doctype html>
+<html>
+<head>
+<script>
+// Closes IE, Edge, Chrome, Brave
+window.onload = function load() {
+  window.open('', '_self', '');
+  window.close();
+};
+</script>
+</head>
+<body>
+  <p>Authentication successful, you can close the browser now.</p>
+  <script>
+    // Needed for Firefox security
+    setTimeout(function() {
+          window.close()
+    }, 5000);
+  </script>
+</body>
+</html>
+"""
+
+
+def init_client():
+    client = hvac.Client(url="https://vault.us1.ddbuild.io")
```

Comment:
We should think of a way where this works for contributors as right now it would fail slowly and always produce a warning. Maybe have telemetry be opt-in, or try to open a socket to test which is very fast?